content/blog/ditch-your-version-manager/index.md

```diff
@@ -0,0 +1,336 @@
+---
+title: Ditch your version manager
+date: "2021-09-17T12:00:00.000Z"
+description: Try Nix and don't look back.
+---
+
+You probably use a variety of tools to manage your Ruby, Node, Python,
+Elixir versions, such as [rvm](http://rvm.io/),
+[rbenv](https://github.com/rbenv/rbenv),
+[nvm](https://github.com/nvm-sh/nvm), or [asdf](https://asdf-vm.com/).
+They all work reasonably well, right?
+
+Well, not quite.
+
+The fundamental issue is that many programs depend on _other_ programs being
+present in the same system. For many years, if you wanted to install
+[nokogiri](https://nokogiri.org/), a popular XML parser written in Ruby,
+you'd have to remember to install `libxml` beforehand. Furthermore, some of
+these programs depend on OS-specific configurations, such as libraries
+shipped with Mac OS or Linux.
+
+It's not uncommon to check out a project and having to install a plethora
+of libraries and tools, and having to hope that the versions that you
+installed are compatible with the ones used by the original authors. Then
+you'd have to figure out a way to install an older version of a specific
+package.
+
+I can hear what you're wondering now: what about
+[Docker](https://www.docker.com/)? The idea of having a container image
+that contains all your dependencies sounds very promising indeed. The issue
+with docker images is that all those dependencies are still intertwined,
+only the whole dependency tree has become *invisible*: for example, if the
+version of Ubuntu you are using has reached its End-Of-Life, you will have
+to manually untangle all the dependencies and update everything in one go.
+If you're working on a large project with many dependencies, please do accept
+my condolences.
+
+My ideal dependency manager would allow me to specify **each and every**
+dependency that is required to work on my projects. It should be easily
+reproducible, declarative and easy to upgrade.
+
+Does it sound too good to be true?
+
+## Welcome to the jungle
+
+Let me introduce you to [Nix](https://nixos.org/).
+
+> Nix is a tool that takes a unique approach to package management and system configuration. Learn how to make reproducible, declarative and reliable systems.
+
+Let's try it out.
+
+```bash
+$ sh <(curl -L https://nixos.org/nix/install) --daemon
+```
+
+```bash
+$ nix-env --version
+nix-env (Nix) 2.3.10
+```
+
+Before we can start on our first project we will need to install
+[direnv](https://direnv.net/), a tool which will let us build
+a custom environment in a specific folder. You'll need to:
+
+- `brew install direnv`
+- [hook direnv into your shell](https://direnv.net/docs/hook.html)
+- restart your shell
+
+Now we're ready and raring to go! 🏄‍♀️
+
+Let's create a new folder:
+
+```bash
+$ mkdir nix_hello_world && cd nix_hello_world
+
+$ hello
+-bash: hello: command not found
+```
+
+As you can see, no `hello` command is available yet.
+
+Great, what I haven't told you is that Nix is also a pure, lazy, functional
```

Comment:
What about: 'Nix configuration is written in "Nix Expression Language" - a pure, lazy, functional programming language'?